src/v/kafka/client/broker.cc

```diff
@@ -58,9 +58,13 @@ ss::future<shared_broker_t> make_broker(
           return ss::make_exception_future<shared_broker_t>(ex);
       })
       .then([&config](shared_broker_t broker) {
-          return do_authenticate(broker, config).then([broker]() {
-              return broker;
-          });
+          return do_authenticate(broker, config)
+            .then([broker]() { return broker; })
+            .handle_exception([broker](std::exception_ptr ex) {
+                return broker->stop().then([ex]() {
```

Comment:
Am I correct in saying that the main issue was the fact that the `transport` instance was never stopped?  Should there be a vassert in `transport`'s (or `base_transport`'s) destructor to catch if `stop()` was not called?

Reply:
The `output_stream` it contains has an assert like that.

We have many objects that have an async `start()`/`stop()` and don't assert in the destructor that it's been stopped if (and only if) it's been started. It would probably have made this a little easier to find.